packages/angular/cli/models/json-schema.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonObject, JsonValue, parseJson } from '@angular-devkit/core';
+import * as jsonSchemaTravese from 'json-schema-traverse';
+import { Option } from './command';
+
+export async function convertSchemaToOptions(schema: string): Promise<Option[]> {
+  const options = await getOptions(schema);
+
+  return options;
+}
+
+function getOptions(schemaText: string, onlyRootProperties = true): Promise<Option[]> {
+  // TODO: refactor promise to an observable then use `.toPromise()`
+  return new Promise((resolve, reject) => {
+    const fullSchema = parseJson(schemaText);
+    const traverseOptions = {};
+    const options: Option[] = [];
+    // tslint:disable-next-line:no-any
+    function postCallback(schema: any,
+                          jsonPointer: string,
+                          rootSchema: string,
+                          parentJsonPointer: string,
+                          parentKeyword: string,
+                          parentSchema: string,
+                          property: string) {
+      if (parentKeyword === 'properties') {
+        let includeOption = true;
+        if (onlyRootProperties && isPropertyNested(jsonPointer)) {
+          includeOption = false;
+        }
+
+        const option: Option = {
+          name: property,
+          // ...schema,
+
+          description: schema.description,
+          type: schema.type,
```

Comment:
This will fail for more complex constraints (`anyOf`/`allOf`/`not`/etc.).  type can potentially be an array as well and is also optional in which case anything is allowed.